src/main/java/com/exactpro/th2/sailfish/utils/filter/NullFilter.java

```diff
@@ -51,10 +48,10 @@ public FilterOperation getOperation() {
 
     @Override
     public ExpressionResult validate(Object value) throws RuntimeException {
-        boolean result = FilterUtils.isNull(value) == acceptNull;
-        return filterSettings.isCheckNullValueAsEmpty()
-                ? ExpressionResult.create(result)
-                : ExpressionResult.create(!result);
+        if (value == NULL_VALUE) {
+            return ExpressionResult.create(checkNullValueAsEmpty == acceptNull);
+        }
+        return ExpressionResult.create(value == null == acceptNull);
```

Comment:
To make code a bit clear I suggest to replace `value == null` with `Objects.isNull(value)`